### PR TITLE
Promote chat_completion in inference guide

### PR DESCRIPTION
**Pull request by Wauplin**
cc @gary149 who suggested it on slack. This PR adds a `chat_completion` example in the `Getting started` section of the guide.

### CODE REVIEW

File: docs/source/en/guides/inference.md [hunk 1]
```diff
@@ -38,13 +38,34 @@ Let's get started with a text-to-image task:
...
+```
+
+In the example above, we initialized an [`InferenceClient`] with the default parameters. The only thing you need to know is the [task](#supported-tasks) you want to perform. By default, the client will connect to the Inference API and select a model to complete the task. In our example, we generated an image from a text prompt. The returned value is a `PIL.Image` object that can be saved to a file. For more details, check out the [`~InferenceClient.text_to_image`] documentation.
+
+Let's now see an example using the chat-completion API. This task uses an LLM to generate a response from a list of messages:
+
+```python
+>>> from huggingface_hub import InferenceClient
+>>> messages = [{"role": "user", "content": "What is the capital of France?"}]
+>>> client = InferenceClient("HuggingFaceH4/zephyr-7b-beta")
```
Comment (gary149) [inline docs/source/en/guides/inference.md:51]: Maybe a bit more popular/up to date:

```suggestion
>>> client = InferenceClient("meta-llama/Meta-Llama-3-8B-Instruct")
```

File: docs/source/en/guides/inference.md [hunk 2]
```diff
@@ -38,13 +38,34 @@ Let's get started with a text-to-image task:
 >>> client = InferenceClient()
 
 >>> image = client.text_to_image("An astronaut riding a horse on the moon.")
->>> image.save("astronaut.png")
+>>> image.save("astronaut.png")  # 'image' is a PIL.Image object
+```
+
+In the example above, we initialized an [`InferenceClient`] with the default parameters. The only thing you need to know is the [task](#supported-tasks) you want to perform. By default, the client will connect to the Inference API and select a model to complete the task. In our example, we generated an image from a text prompt. The returned value is a `PIL.Image` object that can be saved to a file. For more details, check out the [`~InferenceClient.text_to_image`] documentation.
+
+Let's now see an example using the chat-completion API. This task uses an LLM to generate a response from a list of messages:
```
Comment (gary149) [inline docs/source/en/guides/inference.md:46]: Maybe specify what kind of models are compatible with the chat-completion API.

**Review by gary149 — APPROVED** [review]
Nice! cc @merveenoyan maybe a chat completion example + short explanation on https://huggingface.co/tasks/text-generation could be nice too.

File: docs/source/en/guides/inference.md (hunk 2 above)
Comment (gary149) [reply docs/source/en/guides/inference.md:46]: `chat-completion` -> `chat_completion`

Comment (Wauplin) [reply docs/source/en/guides/inference.md:46]: Good call, I have added a link to https://huggingface.co/models?other=conversational&sort=likes

File: docs/source/en/guides/inference.md (hunk 1 above)
Comment (Wauplin) [reply docs/source/en/guides/inference.md]: :+1: 
I've updated the example in docstrings as well (output was quite outdated)

**Pull request merged**